zstd_vendor/patches/0002.patch

```diff
@@ -0,0 +1,44 @@
+diff --git a/lib/legacy/zstd_v01.c b/lib/legacy/zstd_v01.c
+index 1a3aad07..10071134 100644
+--- a/lib/legacy/zstd_v01.c
++++ b/lib/legacy/zstd_v01.c
+@@ -1487,6 +1487,7 @@ static size_t ZSTDv01_decodeLiteralsBlock(void* ctx,
+     BYTE* const oend = ostart + maxDstSize;
+     blockProperties_t litbp;
+
++    memset(&litbp, 0, sizeof(litbp));
```

Comment:
These two warnings:

```bash
/root/rhel_ws/build/zstd_vendor/zstd_vendor-prefix/src/zstd_vendor/lib/legacy/zstd_v01.c: In function ‘ZSTD_decompressBlock’:
/root/rhel_ws/build/zstd_vendor/zstd_vendor-prefix/src/zstd_vendor/lib/legacy/zstd_v01.c:1504:20: warning: ‘litbp.origSize’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 1504 |             size_t rleSize = litbp.origSize;
      |                    ^~~~~~~
/root/rhel_ws/build/zstd_vendor/zstd_vendor-prefix/src/zstd_vendor/lib/legacy/zstd_v01.c:1488:23: note: ‘litbp.origSize’ was declared here
 1488 |     blockProperties_t litbp;
      |                       ^~~~~
/root/rhel_ws/build/zstd_vendor/zstd_vendor-prefix/src/zstd_vendor/lib/legacy/zstd_v01.c:1495:5: warning: ‘*(unsigned int *)((char *)&litbp + offsetof(blockProperties_t, blockType))’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 1495 |     switch(litbp.blockType)
      |     ^~~~~~
/root/rhel_ws/build/zstd_vendor/zstd_vendor-prefix/src/zstd_vendor/lib/legacy/zstd_v01.c:1488:23: note: ‘*(unsigned int *)((char *)&litbp + offsetof(blockProperties_t, blockType))’ was declared here
 1488 |     blockProperties_t litbp;
      |
```

Other solution is replace `memset` with 

```
+    litbp.blockType = 0;
+    litbp.origSize = 0;
```

Reply:
@clalancette As regards
> If it returns an error, then the line if (ZSTDv01_isError(litcSize)) return litcSize; should get us out of here.

I think that clang-tidy is not smart enough to figure out that `(ZSTDv01_isError(litcSize)` will be true for settled `ERROR(srcSize_wrong)` from the ` ZSTDv01_getcBlockSize(..)` at least it seems it doesn't connect this flow.

This is certainly a false positive warning in this case.

BTW if go with the second option
```
 + litbp.blockType = 0;
 + litbp.origSize = 0;
```
It would be more appropriate to init `litbp.blockType = bt_end`. In this case, we will return an error from the switch statement at least.